kafka-connect-s3/src/test/java/io/confluent/connect/s3/TestWithMockedS3.java

```diff
@@ -107,18 +108,20 @@ public void tearDown() throws Exception {
   }
 
   public static Collection<Object> readRecords(String topicsDir, String directory, TopicPartition tp, long startOffset,
-                                               String extension, String zeroPadFormat, String bucketName,
-                                               AmazonS3 s3) throws IOException {
-      String fileKey = FileUtils.fileKeyToCommit(topicsDir, directory, tp, startOffset, extension, zeroPadFormat);
+                                               String extension, String zeroPadFormat, CompressionType compressionType,
+                                               String bucketName, AmazonS3 s3) throws IOException {
+      String fileKey = FileUtils.fileKeyToCommit(topicsDir, directory, tp, startOffset,
+          extension + compressionType.extension, zeroPadFormat);
       if (".avro".equals(extension)) {
         return readRecordsAvro(bucketName, fileKey, s3);
       } else if (".json".equals(extension)) {
-        return readRecordsJson(bucketName, fileKey, s3);
+        return readRecordsJson(bucketName, fileKey, s3, compressionType);
       } else if (".bin".equals(extension)) {
-        return readRecordsByteArray(bucketName, fileKey, s3,
+        return readRecordsByteArray(bucketName, fileKey, s3, compressionType,
             S3SinkConnectorConfig.FORMAT_BYTEARRAY_LINE_SEPARATOR_DEFAULT.getBytes());
       } else if (".SEPARATORjson".equals(extension)) {
```

Comment:
Couple of PRs down the road, and reading this extension doesn't make much sense anymore. Something custom that would seem less like a bug would be slightly better I guess.